Update GallerySetupUtilities.class

hjschiller opened the pull request:
Tried a fresh install and hit the following error:
Fatal error: Uncaught Error: Non-static method GallerySession::compareIdentifiers() cannot be called statically in /home/gallery2/lib/support/GallerySetupUtilities.class:370 Stack trace: #0 /home/gallery2/install/index.php(79): GallerySetupUtilities::startSession() #1 {main} thrown in /home/gallery2/lib/support/GallerySetupUtilities.class on line 370

made proposed change on line 365 and 370 and now hitting:

Warning: A non-numeric value encountered in /home/gallery2/install/steps/SystemChecksStep.class on line 43GalleryUtilities.class

Warning: A non-numeric value encountered in /home/gallery2/install/steps/SystemChecksStep.class on line 43

Warning: A non-numeric value encountered in /home/gallery2/install/steps/SystemChecksStep.class on line 43

Fatal error: Uncaught TypeError: method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given in /home/gallery2/modules/core/classes/GalleryUtilities.class:1216 Stack trace: #0 /home/gallery2/modules/core/classes/GalleryUtilities.class(1216): method_exists() #1 /home/gallery2/modules/core/classes/GalleryUtilities.class(1202): GalleryUtilities::readIndividualManifest() #2 /home/gallery2/install/steps/SystemChecksStep.class(476): GalleryUtilities::readManifest() #3 /home/gallery2/install/steps/SystemChecksStep.class(390): SystemChecksStep->CheckManifest() #4 /home/gallery2/install/index.php(158): SystemChecksStep->processRequest() #5 {main} thrown in /home/gallery2/modules/core/classes/GalleryUtilities.class on line 1216

hjschiller committed:
Update GallerySetupUtilities.class

Tried a fresh install and hit the following error:
Fatal error: Uncaught Error: Non-static method GallerySession::compareIdentifiers() cannot be called statically in /home/gallery2/lib/support/GallerySetupUtilities.class:370 Stack trace: #0 /home/gallery2/install/index.php(79): GallerySetupUtilities::startSession() #1 {main} thrown in /home/gallery2/lib/support/GallerySetupUtilities.class on line 370

gregstoll commented on lib/support/GallerySetupUtilities.class:
```diff
@@ -362,12 +362,12 @@ class GallerySetupUtilities {
 	 * attack.  Either create a clean session, or if we detect that this session is clean,
 	 * sign it in a way that we'll recognize.
 	 */
-	$remoteId = GallerySession::getRemoteIdentifier();
```
I don't think we should need this change - `getRemoteIdentifier()` is already a static method, and I didn't see an error message about this, so I think we can just revert this line.

hjschiller replied:
lgtm.  i'm more comfortable in perforce, so it is not immediately obvious hot to revert, or should I just abandon the change request?

gregstoll replied:
Yeah, I'm not very experienced at GitHubbing :-) but I think the usual practice is just to abandon the pull request (and delete your branch, if you want) - since it hasn't been merged in, no code changes have been made yet, so you can just walk away.

gregstoll commented on lib/support/GallerySetupUtilities.class:
```diff
@@ -362,12 +362,12 @@ class GallerySetupUtilities {
 	$configDir = GallerySetupUtilities::getConfigDir();
 	if (!empty($sessionId) && (
 		    !isset($_SESSION['_remoteId'])
-		    || GallerySession::compareIdentifiers($_SESSION['_remoteId'], $remoteId) == 0
```
Here I think we need to make `compareIdentifiers()` a static method on line 1306 of `modules/core/classes/GallerySession.class`. We'll probably also need to change line 251 of that same file to call this method with the static syntax.

hjschiller replied:
lgtm, happy for you to make the change directly, rather than me bumble around.

gregstoll reviewed (changes requested):
Thanks for taking a stab at this! I don't think this is the right fix, though; I've added some comments to the review. Let me know if you want to try to make them or if you'd like me to.

hjschiller commented:
lgtm. i'm happy to revert (thought its not obvious to me how to do that). 

Feel free to make the updates.  i can happily test.

hjschiller committed:
revert changes to GallerySetupUtilities.class

reverting changes to lib/support/GallerySetupUtilities.class

per:  gregstoll
Here I think we need to make compareIdentifiers() a static method on line 1306 of modules/core/classes/GallerySession.class. We'll probably also need to change line 251 of that same file to call this method with the static syntax.

gregstoll commented:
Per the conversation above, going to fix this another way.
